### PR TITLE
re-implement using oauth

### PR DESCRIPTION
**Pull request by panthershark**
- [x] Update typings from latest API
- [x] Replace cognito auth with oauth
- [ ] Determine a test strategy
- [ ] Update README

### CODE REVIEW

File: src/neris_api_client/config.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    PASSWORD = "password"
+
+class BaseUrl(StrEnum):
+    LOCAL = "http://localhost:8000"
+    DEV = "https://api-dev.neris.fsri.org/v1"
+    TEST = "https://api-test.neris.fsri.org/v1"
+    PROD = "https://api.neris.fsri.org/v1"
+
+
+class Config(BaseModel):
```
Comment (saltaverde) [inline src/neris_api_client/config.py:18]: Might want to put a model validator on this model which looks at `grant_type` and makes sure the correct attributes are non-null e.g. `client_secret` and `client_id` if the grant type is `client_credentials`

File: src/neris_api_client/client.py [hunk 2]
```diff
@@ -57,147 +36,59 @@ def default(self, obj):
...
+                    res = self._session.post(
+                        '/token', 
+                        headers={"Authorization": f"Basic {client_creds}"},
+                        json={"grant_type": GrantType.CLIENT_CREDENTIALS},
+                    )
 
-        self._access_token = res["AuthenticationResult"]["AccessToken"]
-        self._id_token = res["AuthenticationResult"]["IdToken"]
+        elif self.tokens.expires_at <= datetime.now():
+             res = self._session.post(
```
Comment (saltaverde) [inline src/neris_api_client/client.py:73]: What if the refresh token happens to be expired, too? Should the client check the expiration time of the refresh token and start again with the original auth flow? (this is probably an edge case that will get less and less likely the longer the expiration on the refresh token)

File: src/neris_api_client/config.py (hunk 1 above)
Comment (panthershark) [reply src/neris_api_client/config.py:18]: I added a validator in the next commit.

**Review by cweinschenk — APPROVED** [review]
LGTM

**Pull request merged**